includes/Shopping/WooCommerce_Query.php

```diff
@@ -76,12 +78,17 @@ public function get_search( string $search_term ) {
 			[
 				'status'  => 'publish',
 				'limit'   => 100,
-				'orderby' => 'date',
-				'order'   => 'DESC',
 				's'       => $search_term,
+				'orderby' => $orderby,
+				'order'   => $order,
 			]
 		);
 
+		if ( 'price' === $orderby ) {
+			// @todo this only orders products based on wc_get_products previously called
+			$products = wc_products_array_orderby( $products, 'price', $order );
```

Comment:
@swissspidy What if there are more than 100 products that meet the search term. Like you have 1000 products with the name t-shirt. Wouldn't this only sort the top 100?

Reply:
Uhm, yeah, I guess. Seems like that's the only way WooCommerce provides out of the box.

I think on Monday we discussed using meta queries as a workaround

Does something like https://github.com/woocommerce/woocommerce/wiki/wc_get_products-and-WC_Product_Query#adding-custom-parameter-support work? In combination with `'orderby' => 'meta_value',`

Reply:
@spacedmonkey your correct per todo note 

> this only orders products based on wc_get_products previously called


Looks like maybe we can do something with https://woocommerce.wp-a2z.org/oik_api/get_catalog_ordering_args

```php
case 'price':
        $callback = 'DESC' === $order ? 'order_by_price_desc_post_clauses' : 'order_by_price_asc_post_clauses';
        [add_filter](https://core.wp-a2z.org/oik_api/add_filter)( ['posts_clauses'](https://woocommerce.wp-a2z.org/oik_hook/posts_clauses/), array( [$this](https://woocommerce.wp-a2z.org/oik_api/$this::WC_Query), [$callback](https://woocommerce.wp-a2z.org/oik_api/$this::WC_Query) ) );
        break;
```

Also found 

https://stackoverflow.com/questions/47056744/woocommerce-order-by-price-not-working-correctly

```php
/**
 * Customize ordering by price
 */
add_filter('woocommerce_get_catalog_ordering_args', function ($args) {
    $orderby_value = isset($_GET['orderby']) ? wc_clean($_GET['orderby']) : apply_filters('woocommerce_default_catalog_orderby', get_option('woocommerce_default_catalog_orderby'));

    if ('price' == $orderby_value) {
        $args['orderby'] = 'meta_value_num';
        $args['order'] = 'ASC';
        $args['meta_key'] = '_price';
    }

    if ('price-desc' == $orderby_value) {
        $args['orderby'] = 'meta_value_num';
        $args['order'] = 'DESC';
        $args['meta_key'] = '_price';
    }

    return $args;
});

```

Reply:
Updated with this commit --- seems to work 

https://github.com/GoogleForCreators/web-stories-wp/pull/11552/commits/d8fba81ed96fd3525cbfc5c74ef164ff7679c613